src/pages/static/js/settings.js

```diff
@@ -273,9 +316,58 @@ document.addEventListener('DOMContentLoaded', async () => {
   });
 
   // Add change listeners for form elements
-  searchEngine?.addEventListener('change', async (e) => {
-    console.log('Search engine changed:', e.target.value);
-    await saveSettingToBackend('searchEngine', e.target.value);
+   searchEngine?.addEventListener("change", async (e) => {
+    const value = e.target.value;
+    console.log("Search engine changed (UI):", value);
+
+    if (value === "custom") {
+      // Show inline input/modal, but do NOT save the engine yet
+      if (customSearchRow) customSearchRow.style.display = "";
+      // optional UX: prefill from existing template and focus
+      customSearchTemplate?.focus();
+      customSearchTemplate?.select?.();
+      return; // do NOT call settings.set('searchEngine', 'custom') yet
+    }
+
+    // For all non-custom engines, persist immediately and hide the row
+    await saveSettingToBackend("searchEngine", value);
+    if (customSearchRow) customSearchRow.style.display = "none";
+  });
+
+  customSearchTemplate?.addEventListener("input", () => {
+    customSearchMessage.style.display = ""; // show again while editing
+    const state = validateSearchTemplate(customSearchTemplate.value);
+    setTemplateFieldState(customSearchTemplate, customSearchMessage, state);
+  });
+
+  // Save custom search template on Enter
+  customSearchTemplate?.addEventListener("keydown", async (e) => {
```

Comment:
Here, when a user adds a URL as a custom search engine, we should check whether it already exists in the browser’s built-in search engines.
For example, if someone tries to add `https://duckduckgo.com/?q=%s` as a custom template, the UI could show a warning:
"This search engine already exists in the browser."
This prevents duplication and keeps the custom search list clean.